### PR TITLE
Add outer join BBE

### PR DESCRIPTION
**Pull request by SasinduDilshara**

### CODE REVIEW

File: examples/outer-join-clause/outer_join_clause.bal [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Department d2 = {id: 2, name:"Operations"};
+
+    Person[] personList = [p1, p2, p3];
+    Department[] deptList = [d1, d2];
+
+    DeptPerson[] deptPersonList =
+       from var person in personList
+       // The `outer join` clause include all of the records from the left table,
+       // even if there are no matching values for records in the right table.
+       outer join Department dept in deptList
```
Comment (pcnfernando) [inline examples/outer-join-clause/outer_join_clause.bal:34]: Shall we just say 
> The `outer join` clause performs a left outer equijoin. For the records for which there is no matching record on the `deptList`, the resulting record will contain `nil` fields.
wdyt @SasinduDilshara @gimantha ?

Comment (pcnfernando) [inline examples/outer-join-clause/outer_join_clause.bal:34]: Btw, this is will be disallowed with https://github.com/ballerina-platform/ballerina-lang/issues/38223

File: examples/outer-join-clause/outer_join_clause.bal [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    Department d1 = {id: 1, name:"HR"};
+    Department d2 = {id: 2, name:"Operations"};
+
+    Person[] personList = [p1, p2, p3];
+    Department[] deptList = [d1, d2];
+
+    DeptPerson[] deptPersonList =
+       from var person in personList
+       // The `outer join` clause include all of the records from the left table,
```
Comment (praneesha) [inline examples/outer-join-clause/outer_join_clause.bal:32]:
```suggestion
       // The `outer join` clause includes all of the records from the left table
```

File: examples/outer-join-clause/outer_join_clause.bal [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+       // even if there are no matching values for records in the right table.
+       outer join Department dept in deptList
+       on person.id equals dept.id
+       select {
+           fname : person.fname,
+           dept : dept.name
+       };
```
Comment (pcnfernando) [inline examples/outer-join-clause/outer_join_clause.bal:39]:
```suggestion
       outer join var dept in deptList
       on person.id equals dept?.id
       select {
           fname : person.fname,
           dept : dept?.name
       };
```

File: examples/outer-join-clause/outer_join_clause.bal [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    Person p3 = {id: 3, fname: "Ted", lname: "Perera"};
+
+    Department d1 = {id: 1, name:"HR"};
+    Department d2 = {id: 2, name:"Operations"};
+
+    Person[] personList = [p1, p2, p3];
+    Department[] deptList = [d1, d2];
+
+    DeptPerson[] deptPersonList =
+       from var person in personList
```
Comment (pcnfernando) [inline examples/outer-join-clause/outer_join_clause.bal:31]:
```suggestion
       from Person person in personList
```

File: examples/outer-join-clause/outer_join_clause.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Outer join clause
+
+Outer join clause perform `left outer equijoin`. The `outer join` clause include all of the records from the left table, even if there are no matching values for records in the right table. The type to join on must be `anydata`.
```
Comment (praneesha) [inline examples/outer-join-clause/outer_join_clause.md:3]:
```suggestion
the `outer join` clause performs a `left outer equijoin`. The `outer join` clause includes all of the records from the left table even if there are no matching values for records in the right table. The type to join must be `anydata`.
```

File: examples/outer-join-clause/outer_join_clause.metatags [hunk 6]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates outer join clause in Ballerina.
```
Comment (praneesha) [inline examples/outer-join-clause/outer_join_clause.metatags:1]:
```suggestion
description: This BBE demonstrates the outer join clause in Ballerina.
```

File: examples/outer-join-clause/outer_join_clause.out [hunk 7]
```diff
@@ -0,0 +1,2 @@
+$ bal run outer_join_clause.bal
```
Comment (praneesha) [inline examples/outer-join-clause/outer_join_clause.out:1]:
```suggestion
$ bal run outer_join_clause.bal
```
The command should be added in the `.md` file and only the output should be there in the `.out` file.

**Review by praneesha — APPROVED** [review]

**Review by pcnfernando — APPROVED** [review]

**Comment by pcnfernando** [comment]
@SasinduDilshara you'll need to add the BBE to `index.json` after `Join clause`

**Pull request merged**